var/spack/repos/builtin/packages/lua/package.py

```diff
@@ -3,14 +3,219 @@
+                lua_paths.append(os.path.join(d.prefix, self.lua_lib64_dir))
+                lua_paths.append(os.path.join(d.prefix, self.lua_share_dir))
+
+        lua_patterns = []
+        lua_cpatterns = []
+        for p in lua_paths:
+            if os.path.isdir(p):
+                self.append_paths(lua_patterns, lua_cpatterns, p)
+
+        # Always add this package's paths
+        for p in (
+                os.path.join(self.spec.prefix, self.lua_lib_dir),
+                os.path.join(self.spec.prefix, self.lua_lib64_dir),
+                os.path.join(self.spec.prefix, self.lua_share_dir),
+        ):
+            self.append_paths(lua_patterns, lua_cpatterns, p)
+
+        return lua_patterns, lua_cpatterns
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        lua_patterns, lua_cpatterns = self._setup_dependent_env_helper(
+            env, dependent_spec
+        )
+
+        env.prepend_path("LUA_PATH", ";".join(lua_patterns), separator=";")
+        env.prepend_path("LUA_CPATH", ";".join(lua_cpatterns), separator=";")
+
+    def setup_dependent_run_environment(self, env, dependent_spec):
+        # For run time environment set only the path for dependent_spec and
+        # prepend it to LUAPATH
+        lua_patterns, lua_cpatterns = self._setup_dependent_env_helper(
+            env, dependent_spec
+        )
+
+        if dependent_spec.package.extends(self.spec):
+            env.prepend_path("LUA_PATH", ";".join(lua_patterns), separator=";")
+            env.prepend_path("LUA_CPATH", ";".join(lua_cpatterns), separator=";")
+
+    def setup_run_environment(self, env):
+        env.prepend_path(
+            "LUA_PATH",
+            os.path.join(self.spec.prefix, self.lua_share_dir, "?.lua"),
+            separator=";",
+        )
+        env.prepend_path(
+            "LUA_PATH",
+            os.path.join(self.spec.prefix, self.lua_share_dir, "?", "init.lua"),
+            separator=";",
+        )
+        env.prepend_path(
+            "LUA_PATH",
+            os.path.join(self.spec.prefix, self.lua_lib_dir, "?.lua"),
+            separator=";",
+        )
+        env.prepend_path(
+            "LUA_PATH",
+            os.path.join(self.spec.prefix, self.lua_lib_dir, "?", "init.lua"),
+            separator=";",
+        )
+        env.prepend_path(
+            "LUA_CPATH",
+            os.path.join(self.spec.prefix, self.lua_lib_dir, "?.so"),
+            separator=";",
+        )
+
+    @property
+    def lua(self):
+        return Executable(join_path(self.spec.prefix.bin, 'lua'))
+
+    @property
+    def luarocks(self):
+        return Executable(join_path(self.spec.prefix.bin, 'luarocks'))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        """
+        Called before lua modules's install() methods.
+
+        In most cases, extensions will only need to have two lines::
+
+        luarocks('--tree=' + prefix, 'install', rock_spec_path)
```

Comment:
I think this line needs to be indented for the double colon above to work. Not that we generate docs for packages anyway.